app/controllers/actions/TaxYearAction.scala

```diff
@@ -32,13 +32,13 @@ class TaxYearAction @Inject()(taxYear: Int)
   lazy val logger: Logger = Logger.apply(this.getClass)
 
   override def refine[A](request: IdentifierRequest[A]): Future[Either[Result, IdentifierRequest[A]]] = {
```

Comment:
I feel this could be made more simple and readable, something like the following maybe?
`override def refine[A](request: IdentifierRequest[A]): Future[Either[Result, IdentifierRequest[A]]] = {

    val isValidYear = request.session.get("validTaxYears").map(_.split(",")) match {
      case Some(value) => value.map(_.toInt).contains(taxYear)
      case None => false
    }

    if (isValidYear) {
      Future.successful(Right(IdentifierRequest(request.request, request.mtdItId, request.isAgent)))
    } else {
      logger.info(s"[TaxYearAction][refine] Invalid tax year, redirecting to error page")
      // todo should redirect to where user selects taxYear
      Future.successful(Left(Redirect(controllers.routes.IncorrectTaxYearErrorPageController.onPageLoad(taxYear))))
    }
  }`

Reply:
Yep agreed, I have made this change in the next commit